src/state/types.ts

```diff
@@ -58,8 +58,23 @@ export interface FarmsState {
   userDataLoaded: boolean
 }
 
+export interface CakeVault {
+  totalShares?: string
```

Comment:
When the balance strings are typed as BigNumbers, it throws an error after serialising the BigNumber with .toJSON():

```
Type 'string' is not assignable to type 'BigNumber'
```

Hence typing them as strings within redux.

Reply:
Farm & Pool state doesn't seem to need to type these as BigNumbers... although I'm struggling to see how they've got around it.

Generally looking for feedback on serialisation of these contract responses, example data:

`totalShares: "18574889782429049977066502"`

Reply:
Farms having values as BigNumber is just a mistake, and it wasn't throwing errors just cause TS didn't actually check them after bunch of hops through helper functions/hooks. It was actually fixed in some places to use string. So everything's fine here 👍

Reply:
Ah excellent news.

Reply:
If the number is a balance that needs decimal conversion it should be stored as a big number. In Redux `bn.toJSON()` and out of redux `new BigNumber(bn)`. It is a more overhead I admit but I think much better if we consistently do that.